src/librustc_errors/registry.rs

```diff
@@ -1,16 +1,40 @@
+use crate::registry::RegistryResult::HasLongDescription;
 use rustc_data_structures::fx::FxHashMap;
 
+pub enum RegistryResult {
+    InvalidErrorCode,
+    NoLongDescription,
+    HasLongDescription(&'static str),
+}
+
+impl RegistryResult {
+    pub fn has_value(&self) -> bool {
+        match self {
+            HasLongDescription(_s) => true,
+            _ => false,
+        }
+    }
+}
+
 #[derive(Clone)]
 pub struct Registry {
-    descriptions: FxHashMap<&'static str, &'static str>,
+    long_descriptions: FxHashMap<&'static str, Option<&'static str>>,
 }
 
 impl Registry {
-    pub fn new(descriptions: &[(&'static str, &'static str)]) -> Registry {
-        Registry { descriptions: descriptions.iter().cloned().collect() }
+    pub fn new(long_descriptions: &[(&'static str, Option<&'static str>)]) -> Registry {
+        Registry { long_descriptions: long_descriptions.iter().cloned().collect() }
     }
 
-    pub fn find_description(&self, code: &str) -> Option<&'static str> {
-        self.descriptions.get(code).cloned()
+    pub fn find_description(&self, code: &str) -> RegistryResult {
```

Comment:
Looking at the use sites, it looks like most of the time (well, 2 out of 3?) expect to be passing in a valid code, and we should panic if they don't.

I think we'll want the following API:

```rust

#[derive(Debug)]
struct InvalidErrorCode;

/// This will panic if an invalid error code is passed in
pub fn find_description(&self, code: &str) -> Option<&'static str> {
    self.try_find_description(code).unwrap()
}

pub fn try_find_description(&self, code: &str) -> Result<Option<&'static str>, InvalidErrorCode> {
    // ...
}
```